Bottom: 4bbcb14e0681aad0a350fedd37f6fa80dd36ffe4
Top:    8b7f20545a233ec930ffd0b00fae0fe6bd660a5e
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2014-06-26 21:12:40 +0100

Add support for a two soldermask layers, defining regions of removed mask

File-format wise, these work similarly to how silk layers are supported now:

Layer (1 to n "[NAME]")     # "ordinary" design layers
Layer (n+1 "silk")          # top side
Layer (n+2 "silk")          # bottom side
Layer (n+3 "soldermask")    # top side
Layer (n+4 "soldermask")    # bottom side


---

diff --git a/src/action.c b/src/action.c
index 7a797f3..d91b9f5 100644
--- a/src/action.c
+++ b/src/action.c
@@ -1386,7 +1386,9 @@ NotifyMode (void)
 		int flag = CLEARLINEFLAG;
 
 		if (GetLayerGroupNumberByNumber (INDEXOFCURRENT) ==
-		    GetLayerGroupNumberByNumber (bottom_silk_layer))
+		    GetLayerGroupNumberByNumber (bottom_silk_layer) ||
+		    GetLayerGroupNumberByNumber (INDEXOFCURRENT) ==
+		    GetLayerGroupNumberByNumber (bottom_soldermask_layer))
 		  flag |= ONSOLDERFLAG;
 		if ((text = CreateNewText (CURRENT, &PCB->Font, Note.X,
 					   Note.Y, 0, Settings.TextScale,
diff --git a/src/data.h b/src/data.h
index 06daf0c..470933b 100644
--- a/src/data.h
+++ b/src/data.h
@@ -49,6 +49,8 @@ extern PCBType *PCB;
 #define max_copper_layer (PCB->Data->LayerN)
 #define bottom_silk_layer (max_copper_layer + BOTTOM_LAYER)
 #define top_silk_layer (max_copper_layer + TOP_LAYER)
+#define bottom_soldermask_layer (max_copper_layer + BOTTOM_SOLDERMASK_LAYER)
+#define top_soldermask_layer (max_copper_layer + TOP_SOLDERMASK_LAYER)
 
 extern SettingType Settings;
